### PR TITLE
Add support for Notifications

### PR DESCRIPTION
**Pull request by chrisndodge**

### CODE REVIEW

**Comment by chrisndodge** [comment]
@martynjames and here are the changes I'm working on for the Group Projects to fire notifications. This is still WIP.

**Comment by chrisndodge** [comment]
@martynjames thanks for the comments on the Apros PR. Just bringing this also to your attention.

File: group_project/group_project.py [hunk 1]
```diff
@@ -665,11 +673,95 @@ def upload_submission(self, request, suffix=''):
...
                         }
                     )
+                    at_least_one_success = True
                 except Exception as save_record_error:
                     original_message = save_record_error.message if hasattr(save_record_error, "message") else ""
                     save_record_error.message = _("Error recording file information {} - {}").format(uf.file.name, original_message)
                     raise
 
+            if at_least_one_success:
+                # See if the xBlock Notification Service is available, and - if so -
```
Comment (martynjames) [inline group_project/group_project.py]: This is a bit long, probably want to move it into a separate method

Comment (chrisndodge) [inline group_project/group_project.py]: done, moved it out into it's own method

**Comment by martynjames** [comment]
This all looks good to me! Let me know when no longer WIP, and I'll run through a test pass with this code in place

**Comment by chrisndodge** [comment]
@martynjames FYI just commit changeset to support asynchronous due-date notifications.

**Comment by chrisndodge** [comment]
HI @martynjames just following up on your earlier comment to get a heads-up when we're gearing this up to merge onto master. We'd like to do this tomorrow, to keep on the release train schedule.

**Comment by martynjames** [comment]
Code looks good to me - :+1: - still need to run some tests

**Pull request merged**